internify/src/components/molecules/TechStack.js

```diff
@@ -1,80 +1,224 @@
+      jobData.languages.push(event.target.name);
+    } else {
+      if (jobData.languages.includes(event.target.name)) {
+        jobData.languages = jobData.languages.filter(
+          (obj) => obj !== event.target.name
+        );
+      }
+    }
+  };
+
 
   function handleFrameworksChange(event) {
-    jobData.frameworks.push(event.target.name);
-    handleChange({ ...jobData, frameworks: jobData.frameworks });
-  }
+    setFrameState({...frameState, [event.target.name]: event.target.checked});
+    if (event.target.checked) {
+      jobData.frameworks.push(event.target.name);
+    } else {
+      if (jobData.frameworks.includes(event.target.name)) {
+        jobData.frameworks = jobData.frameworks.filter(
+          (obj) => obj !== event.target.name
+        );
+      }
+    }
+  };
+
 
   function handleWorkToolsChange(event) {
-    jobData.tools.push(event.target.name);
-    handleChange({ ...jobData, tools: jobData.tools });
-  }
+    setToolsState({...toolsState, [event.target.name]: event.target.checked});
+    if (event.target.checked) {
+      jobData.tools.push(event.target.name);
+    } else {
+      if (jobData.tools.includes(event.target.name)) {
+        jobData.tools = jobData.tools.filter(
+          (obj) => obj !== event.target.name
+        );
+      }
+    }
+  };
 
   function handleConceptsChange(event) {
-    jobData.concepts.push(event.target.name);
-    handleChange({ ...jobData, concepts: jobData.concepts });
-  }
+    setConceptsState({...conceptsState, [event.target.name]: event.target.checked});
+    if (event.target.checked) {
+      jobData.concepts.push(event.target.name);
+    } else {
+      if (jobData.concepts.includes(event.target.name)) {
+        jobData.concepts = jobData.concepts.filter(
+          (obj) => obj !== event.target.name
+        );
+      }
+    }
+  };
+
 
   return (
     <React.Fragment>
       <h2>Tech stack</h2>
       <div className="tech_stack_container">
         <h3>Coding Languages*</h3>
         <div className="tech_stack_checkbox_container">
-          {languages.map((x) => {
-            return (
-              <CheckBox key={x} name={x} label={x} onChange={handleLanguageChange} />
-            );
-          })}
+          <CheckBox
+          key={"Java"}
+          name={"Java"}
+          label={"Java"}
+          onChange={handleLanguageChange}
+          checked={langState.Java}
+          />
+          <CheckBox 
+          key={"JavaScript"}
+          name={"JavaScript"}
+          label={"JavaScript"}
+          onChange={handleLanguageChange}
+          checked={langState.JavaScript}
+          />
+          <CheckBox 
+          key={"Cpp"}
+          name={"Cpp"}
+          label={"C++"}
+          onChange={handleLanguageChange}
+          checked={langState.Cpp}
+          />
+          <CheckBox 
+          key={"C"}
+          name={"C"}
+          label={"C"}
+          onChange={handleLanguageChange}
+          checked={langState.C}
+          />
```

Comment:
can we try to figure out a way to keep the old format and still update the state? checked={langState[x]}? The same applies for the other changes.

Reply:
reverted back to old format - to my surprised - it (mostly) worked! the only change is that we need to have dedicated handleChange functions for each different tech stack category, else there will be interesting output on the redux window.